docs/guides/src/main/server/configuration-provider.adoc

```diff
@@ -0,0 +1,89 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+<#import "/templates/options.adoc" as opts>
+
+<@tmpl.guide
+title="Configuring providers"
+summary="Understand how to configure providers">
+
+The server is built with extensibility in mind and for that it provides a number of Service Provider Interfaces or SPIs, each one
+responsible for providing a specific capability to the server. In this guide, you are going to understand the core concepts around
+the configuration of SPIs and their respective providers.
+
+After reading this guide, you should be able to use the concepts and the steps herein explained to install, uninstall, enable, disable, and configure
+any provider, including those you have implemented to extend the server capabilities in order to better fulfill your requirements.
+
+== Configuration option format
+
+Providers can be configured by using a specific configuration format. The format consists of:
+
+```
+spi-<spi>-<provider-id>-<property>=<value>
+```
+
+The `<spi>` is the name of the SPI you want to configure.
+
+The `<provider-id>` is the id of the provider you want to configure. This is the id set to the corresponding provider factory implementation.
+
+The `<property>` is the actual name of the property you want to set for a given provider.
+
+All those names (for spi, provider, and property) should be in lower case and if the name is in camel-case such as `myKeycloakProvider`, it should include slashes (`-`) before upper-case letters as follows: `my-keycloak-provider`.
+
+Taking the `HttpClientSpi` SPI as an example, the name of the SPI is `connectionsHttpClient` and one of the provider implementations available is named `default`. In order to set the `connectionPoolSize` property you would use a configuration option as follows:
+
+```
+spi-connections-http-client-default-connection-pool-size=10
+```
+
+== Setting a provider configuration option
+
+The configuration options you set to a provider can be provided when starting the server. The values are available to the providers
+when at the init phase of the provider lifecycle.
+
+If using the CLI, you should be able to set these options as follows:
```

Comment:
_The values are available to the providers
when at the init phase of the provider lifecycle_ -- I had to read this a few times to understand what you meant here. I'd leave this out as it's more a concern when writing custom providers, and should rather be mentioned in the extending the server guide than here.

Could probably also simplify this whole section to something like:

```suggestion
Provider configuration options are provided when starting the server, as shown in the following example:
```